### ISSUE TITLE
Text files with LaTeX extensions do not open in the text editor.

### ISSUE THREAD

**Issue by SentToDevNull** [nextcloud/files_texteditor#24]
Filename extensions include: .tex, .aux, .sty, .bib, .toc, .nav, .cls, .dev, .lof, .bst, .bbx, .lot, .cbx, .bbl, .blg, and .fd

Instead of having a list of common text filename extensions that can be opened, perhaps a better way to determine whether or not to open files is by using magic numbers. 

Source for extensions: https://tex.stackexchange.com/questions/7770/file-extensions-related-to-latex-etc

**Comment by nickvergessen** [comment]
What are the mimetypes of those files?

Can you upload some samples for me here, so I can test it and add them to 
https://github.com/nextcloud/files_texteditor/blob/master/js/editor.js#L240-L252

**Comment by SentToDevNull** [comment]
Here are the mimetypes:
```
text/x-bibtex
text/x-tex
text/x-log
text/plain
application/x-cdrdao-toc
```
Here is a link to sample files with those mimetypes:
[(https://www.lukasyoder.com/nextcloud/index.php/s/QcX3TuDSG2fOJfE)]

**Comment by Spartachetto** [comment]
Just a quick remark: the functionality of the issue **Manage the file extensions associated with texteditor** (#15), would allow the same user (or, maybe better, admin) to perform this operation

**Comment by SentToDevNull** [comment]
Agreed. Seems like that would be a pretty nice feature to have in Nextcloud 12.

**Comment by ruedigerkupper** [comment]
@SentToDevNull:
Your MIME type for the .toc file is wrong. The one you mention (application/x-cdrdao-toc) is for CD-table-of-contents. It's used by the popular „cdrdao“ tool for creating CD images.
LaTeX also creates .toc files (for table of contents of written documents). They have, like all LaTeX files, MIME type „text/plain“.

**Comment by ruedigerkupper** [comment]
@nickvergessen:
All LaTeX files are plain text (originally ASCII, but nowadays usually UTF-8 encoded). That is one feature that made and makes LaTeX so beautiful and long lasting: There is no dedicated editor for LaTeX, just use your favorite text editor, be it Emacs, gedit or vi.
So you would indeed expect that LaTeX files open in your app.
